### PR TITLE
Test result bundle destination

### PR DESCRIPTION
**Pull request by olejnjak**
Resolves https://github.com/tuist/tuist/issues/2186

### Short description 📝

For various use cases it is useful to have the XCResult bundle available when running tests. This PR adds support of `--result-bundle-path` parameter to `TestCommand` which is then forwarded to `xcodebuild`.

I am not entirely sure how it will cope with the automatic caching feature. If I run `tuist test --result-bundle-path Project.xcresult` it will be fine. If I add new framework with a single test and run it again, I would get result bundle with that single test. And such case might be even more interesting on CI as two subsequent jobs would get different result bundles, first one will contain all tests and the other one will get nothing as nothing in the project had changed. Probably introducing some argument like `--skip-cache`/`--no-cache` and maybe even skipping it automatically when `--result-bundle-path` is used would be the way the go? 

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: projects/docs/docs/commands/test.md [hunk 1]
```diff
@@ -50,11 +50,12 @@ One of the benefits of using Tuist over other automation tools is that developer
...
+| ---------------------- | ----- | ------------------------------------------------------------------- | ----------------- | -------- |
+| `--clean`              | n/a   | `When passed, it cleans the project before testing it.`             | False             | No       |
+| `--path`               | `-p`  | `The path to the directory that contains the project to be tested.` | Current directory | No       |
+| `--device`             | `-d`  | `Test on a specific device.`                                        |                   | No       |
+| `--os`                 | `-o`  | `Test with a specific version of the OS.`                           |                   | No       |
+| `--configuration`      | `-C`  | `The configuration to be used when building the scheme.`            |                   | No       |
+| `--skip-ui-tests`      | n/a   | `When passed, it skips testing UI Tests targets.`                   | False             | No       |
+| `--result-bundle-path` | n/a   | `Path where test result bundle will be saved`                       |                   | No       |
```
Comment (pepibumur) [inline projects/docs/docs/commands/test.md:61]: What about adding a short version for the argument? `-T`?

File: projects/docs/docs/commands/test.md [hunk 2]
```diff
@@ -50,11 +50,12 @@ One of the benefits of using Tuist over other automation tools is that developer
...
-| `--path`            | `-p`  | `The path to the directory that contains the project to be tested.` | Current directory | No       |
-| `--device`          | `-d`  | `Test on a specific device.`                                        |                   | No       |
-| `--os`              | `-o`  | `Test with a specific version of the OS.`                           |                   | No       |
-| `--configuration`   | `-C`  | `The configuration to be used when building the scheme.`            |                   | No       |
-| `--skip-ui-tests`   | n/a  | `When passed, it skips testing UI Tests targets.`                   | False             | No       |
+| Argument               | Short | Description                                                         | Default           | Required |
+| ---------------------- | ----- | ------------------------------------------------------------------- | ----------------- | -------- |
+| `--clean`              | n/a   | `When passed, it cleans the project before testing it.`             | False             | No       |
```
Comment (pepibumur) [inline projects/docs/docs/commands/test.md:55]: Same here, I thin we can make `-c` the short version for this one. It's not related to your PR, but we can take the opportunity and extend it.

**Review by pepibumur — APPROVED** [review]
This is great @olejnjak, thanks for adding support for this. I left some nitpick comments. Once they are addressed, the git conflicts are resolved, and CI is green I think we can merge this PR.

**Comment by olejnjak** [comment]
> This is great @olejnjak, thanks for adding support for this. I left some nitpick comments. Once they are addressed, the git conflicts are resolved, and CI is green I think we can merge this PR.

Should be ready now. What do you think about the `--no-cache` argument as mentioned in the PR description?

**Review by fortmarek — APPROVED** [review]
Thanks @olejnjak for the contribution!

The PR as-is is good to merge IMHO, I'd just like to discuss a bit more the implications for test caching.

> even skipping it automatically when --result-bundle-path is used would be the way the go?

I'm not sure about this, if tuist skips some tests and I pass `--result-bundle-path`, I just might not be interested in those skipped tests and might wanna just inspect tests that _did_ run? 

If you want all the tests to be run, you can run `tuist clean` before `tuist test`. But `--skip-cache` might be a more granular approach, so I'm inclined to add that 👍 (however, that is out of scope for this PR)

**Comment by fortmarek** [comment]
Also note that you'll have to wait for [this](https://github.com/tuist/tuist/pull/3193) PR that should fix the checks failing on `main` and then do a rebase, please 🙏

**Pull request merged**